File: data-pipeline-flink/dp-telemetry-reader/src/main/java/org/sunbird/dp/reader/NullParent.java
package org.sunbird.dp.reader;

import org.sunbird.dp.util.Logger;

import java.text.MessageFormat;

// import org.ekstep.ep.samza.core.Logger;

class NullParent implements ParentType {
    static Logger LOGGER = new Logger(NullParent.class);
    Object parent;
    String childKey;

    NullParent(Object parent, String childKey) {
        this.parent = parent;
        this.childKey = childKey;
    }

    @Override
    public <T> T readChild() {
        LOGGER.warn(null, MessageFormat.format("NULL PARENT READ CHILD INVOKED FOR PARENT: {0}, CHILD KEY: {1}", parent, childKey));
        return null;
    }

    @Override
    public void addChild(Object value) {
        LOGGER.warn(null, MessageFormat.format("NULL PARENT ADD CHILD INVOKED FOR PARENT: {0}, CHILD KEY: {1}", parent, childKey));
    }
}

File: data-pipeline-flink/dp-core/src/main/scala/org/sunbird/dp/reader/NullParent.scala
package org.sunbird.dp.reader

import java.text.MessageFormat


// import org.ekstep.ep.samza.core.Logger;

object NullParent {
  //private[reader] val LOGGER = new Logger(classOf[NullParent])
}

class NullParent private[reader](var parent: Any, var childKey: String) extends ParentType {
  override def readChild[T]: Option[T] = {
    //NullParent.LOGGER.warn(null, MessageFormat.format("NULL PARENT READ CHILD INVOKED FOR PARENT: {0}, CHILD KEY: {1}", parent, childKey))
    Some(null.asInstanceOf[T])
  }

  override def addChild(value: Any): Unit = {
   // NullParent.LOGGER.warn(null, MessageFormat.format("NULL PARENT ADD CHILD INVOKED FOR PARENT: {0}, CHILD KEY: {1}", parent, childKey))
  }
}
